src/parser/mod.rs

```diff
@@ -6210,6 +6210,20 @@ impl<'a> Parser<'a> {
         } else {
             let name = self.parse_object_name()?;
 
+            let partitions = if dialect_of!(self is MySqlDialect)
+                && self.parse_keyword(Keyword::PARTITION)
+            {
+                let mut partitions = self.parse_comma_separated(|p| p.parse_tuple(true, false))?;
```

Comment:
As written I think this only checks the first partition for having more than one tuple element (what about values after index 0?)

I think if you used `parse_expr` here this code would be significantly simpler and there would be no need for error checking

```suggestion
                let mut partitions = self.parse_comma_separated(Parser::parse_expr)?;
```

Reply:
Returns an error if there are multiple tuples.